packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+  /// transformation back to its starting position.
+  ///
+  /// ```dart
+  /// final TransformationController _transformationController = TransformationController();
+  /// Animation<Matrix4> _animationReset;
+  /// AnimationController _controllerReset;
+  ///
+  /// void _onAnimateReset() {
+  ///   setState(() {
+  ///     _transformationController.value = _animationReset.value;
+  ///   });
+  ///   if (!_controllerReset.isAnimating) {
+  ///     _animationReset?.removeListener(_onAnimateReset);
+  ///     _animationReset = null;
+  ///     _controllerReset.reset();
+  ///   }
+  /// }
+  ///
+  /// void _animateResetInitialize() {
+  ///   _controllerReset.reset();
+  ///   _animationReset = Matrix4Tween(
+  ///     begin: _transformationController.value,
+  ///     end: Matrix4.identity(),
+  ///   ).animate(_controllerReset);
+  ///   _controllerReset.duration = const Duration(milliseconds: 400);
+  ///   _animationReset.addListener(_onAnimateReset);
+  ///   _controllerReset.forward();
+  /// }
+  ///
+  /// // Stop a running reset to home transform animation.
+  /// void _animateResetStop() {
+  ///   _controllerReset.stop();
+  ///   _animationReset?.removeListener(_onAnimateReset);
+  ///   _animationReset = null;
+  ///   _controllerReset.reset();
+  /// }
+  ///
+  /// void _onInteractionStart(ScaleStartDetails details) {
+  ///   // If the user tries to cause a transformation while the reset animation is
+  ///   // running, cancel the reset animation.
+  ///   if (_controllerReset.status == AnimationStatus.forward) {
+  ///     _animateResetStop();
+  ///   }
+  /// }
+  ///
+  /// IconButton get resetButton {
+  ///   return IconButton(
+  ///     onPressed: () {
+  ///       setState(() {
+  ///         _animateResetInitialize();
+  ///       });
+  ///     },
+  ///     tooltip: 'Reset',
+  ///     color: Theme.of(context).colorScheme.surface,
+  ///     icon: const Icon(Icons.replay),
+  ///   );
+  /// }
+  ///
+  /// @override
+  /// void initState() {
+  ///   super.initState();
+  ///   _controllerReset = AnimationController(
+  ///     vsync: this,
+  ///   );
+  /// }
+  ///
+  /// @override
+  /// void dispose() {
+  ///   _controllerReset.dispose();
+  ///   super.dispose();
+  /// }
+  ///
+  /// @override
+  /// Widget build(BuildContext context) {
+  ///   return Scaffold(
+  ///     backgroundColor: Theme.of(context).colorScheme.primary,
+  ///     appBar: AppBar(
+  ///       automaticallyImplyLeading: false,
+  ///       title: const Text('Controller demo'),
+  ///     ),
+  ///     body: Center(
+  ///       child: InteractiveViewer(
+  ///         boundaryMargin: EdgeInsets.all(double.infinity),
+  ///         disableRotation: true,
+  ///         transformationController: _transformationController,
+  ///         minScale: 0.1,
+  ///         maxScale: 1.0,
+  ///         onInteractionStart: _onInteractionStart,
+  ///         child: Container(
+  ///           color: Colors.pink,
```

Comment:
Would this demo be easier to see if the container's background was a gradient?

Reply:
Well even if it's not easier to see it will definitely be cooler 🔥